src/sidebar/components/ShareDialog/FileInput.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  FileGenericIcon,
+  IconButton,
+  Input,
+  InputGroup,
+} from '@hypothesis/frontend-shared';
+import type { MutableRef } from 'preact/hooks';
+import { useId } from 'preact/hooks';
+
+export type FileInputProps = {
+  onFileSelected: (file: File) => void;
+  fileInputRef: MutableRef<HTMLInputElement | null>;
```

Comment:
> Edit: It looks to me like this is a test seam? See my notes in the tests about possible approaches here.

That's correct. However, I couldn't set a "default" value because (p)react doesn't like conditional hooks 😅 

Ideally I would like to do something like `ref ?? useRef()`, but that is not possible.

Reply:
The ability to make refs optional was one of the motivations for the `useSyncedRef` hook in the `@hypothesis/frontend-shared` package. The other option is to use a function ref like so:

```js
<input ref={el => if (ref) { ref.current = el }}/>
```